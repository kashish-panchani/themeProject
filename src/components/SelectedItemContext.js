import React, { createContext, useContext, useRef, useState } from 'react';
const SelectedItemContext = createContext();
export const useSelectedItem = () => useContext(SelectedItemContext);
export const SelectedItemProvider = ({ children }) => {
  const [selectedItem, setSelectedItem] = useState([]);
  const desktopinputRef= useRef();
  const mobileInputRef=useRef();
  const [isSearchBarVisible, setSearchBarVisible] = useState(false);
  
  return (
    <SelectedItemContext.Provider value={{ selectedItem, setSelectedItem ,desktopinputRef,isSearchBarVisible,setSearchBarVisible,mobileInputRef}}>
      {children}
    </SelectedItemContext.Provider>
  );
};
