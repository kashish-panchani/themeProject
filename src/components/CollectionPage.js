import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { cardItems } from '../constants/home';
import { collection } from '../constants/collectionpage';
import qrCode from "../images/qrcode.webp";
import { useSelectedItem } from './SelectedItemContext';
import { ReactComponent as OpenInNewTabIcon } from "../images/opennewtab.svg";
const CollectionPage = () => {
  const { setSelectedItem } = useSelectedItem();
  const [hoveredItemId, setHoveredItemId] = useState(null);
  const { text } = useParams();
  const selectedItem = cardItems.find((item) => item.text[0] == text);
  useEffect(() => {
    window.scrollTo(0, 0);
    setSelectedItem(selectedItem);
  }, []);
  return (
    <>
      <div style={{ backgroundColor: selectedItem ? selectedItem.bannerbackground : "" }} >
        <div className='mx-0 sm:mx-auto container'>
          {selectedItem && (
            <div style={{ fontFamily: selectedItem.fontFamily }}>
              <h3 className='text-center text-base md:text-2xl py-5' style={{ color: selectedItem.headerBackground || "black" }}>{selectedItem.text}</h3>
            </div>
          )}
          <div className='flex items-center justify-between max-w-xs sm:max-w-3xl md:max-w-4xl w-full mx-auto'>
            <div className='gap-2 flex items-center'>
              {selectedItem && (
                <div
                  style={{ backgroundColor: selectedItem.backgroundColor }}
                  className='flex items-center justify-center rounded-full w-20 h-20 sm:h-24 sm:w-24 md:h-28 md:w-28'
                >
                  <img
                    src={selectedItem.icon}
                    alt={selectedItem.alt}
                    className='opacity-30 w-10 h-10 sm:w-12 sm:h-12 md:w-16 md:h-16'
                  />
                </div>
              )}
              <div className='w-4/6 sm:w-5/6 flex flex-col mb-2 gap-2'>
                <div className='text-xs sm:text-lg uppercase font-medium'>
                  {collection.collection_name}
                </div>
                <div className='text-xs sm:text-sm text-ellipsis line-clamp-2 md:line-clamp-3'>
                  {collection.description}
                </div>
              </div>
            </div>
            <div className='w-96 ml-2 sm:w-52 relative'>
              <img
                src={qrCode}
                alt='collection'
                className='w-full h-full object-contain'
                loading='lazy'
              />
            </div>
          </div>
          <div className='grid grid-cols-2 sm:grid-cols-3 2xl:grid-cols-3 gap-2.5 md:gap-4 py-2 justify-center items-center max-w-80 sm:max-w-4xl xl:max-w-4xl mx-auto'>
            <div className='col-span-2 sm:col-span-2 sm:row-span-2 flex-grow'>
              <img
                className='object-cover rounded-xl sm:h-[423px] sm:w-[425px] md:h-[528px] md:w-[528px] lg:h-[592px] lg:w-[592px]'
                alt='coverimg'
                src={collection.cover_image}
              />
            </div>
            {collection?.product_lists.map((el) => (
              <div
                key={el.id}
                id="product_card"
                className='relative collection_card rounded-xl overflow-hidden'
                data-testid="collection"
                onMouseOver={() => setHoveredItemId(el.id)}
                onMouseLeave={() => setHoveredItemId(null)}
                style={
                  hoveredItemId === el.id
                    ? { '--hover-bg-color': selectedItem.headerBackground }
                    : {}
                }
              >
                <div className='border rounded-xl'>
                  <img
                    src={el.image}
                    alt="collection_image"
                    data-testid="colletionImage"
                    className='object-cover w-60 h-40 sm:w-52 sm:h-52 md:w-64 md:h-64 lg:w-72 lg:h-72'
                  />
                </div>
                <div
                  className='flex absolute top-0 overflow-hidden px-4 h-16 w-full product_card_head'
                  title={collection.collection_name}
                >
                  <div
                    className={`inline-block text-ellipsis whitespace-nowrap w-64 text-white font-bold pt-2 overflow-hidden m-0 cursor-pointer z-10 sm:z-1 
                        ${hoveredItemId === el.id ? 'hovered' : ''}
                        `
                    }
                  >
                    {collection.collection_name}
                  </div>
                </div>
                <div className='flex justify-center items-center text-center absolute text-white text-lg top-0 bottom-0 h-overflow-hidden px-4 mx-auto w-full z-10'>
                  {hoveredItemId === el.id && (
                    <div className=''>
                      <div className='flex justify-center items-center'>
                        <div className='font-bold text-xl capitalize '>buy now</div>
                        <OpenInNewTabIcon className='text-white ml-1 text-xs w-5 h-5 z-10 ' />
                      </div>
                      <div className='text-sm'>From amazonfashion</div>
                    </div>
                  )}
                </div>
                <div
                  className='flex justify-between items-center absolute bottom-0 overflow-hidden p-4 h-16 w-full product_card_footer pt-7'
                  style={{
                    display: hoveredItemId === el.id && 'flex',
                  }}
                >
                  {hoveredItemId === el.id ? (
                    <div className='flex justify-center'>
                      <button
                        className='border uppercase text-white text-xs font-bold px-7 py-2 mx-9 rounded-xl z-10'
                      >
                        add to Collection
                      </button>
                    </div>
                  ) : (
                    <>
                      <p className='whitespace-nowrap w-64 text-white font-bold'>â‚¹{el.price}</p>
                      <p className='whitespace-nowrap text-white font-bold'>{el.offer}</p>
                    </>
                  )}
                </div>
              </div>
            ))}
          </div>
        </div>
      </div>
    </>
  );
}

export default CollectionPage;

