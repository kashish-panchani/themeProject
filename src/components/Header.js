import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useSelectedItem } from './SelectedItemContext';
const { handleCollection } = require('../utils/utils')
const Header = () => {
  const [isDropdownOpen, setDropdownOpen] = useState(false);
  const { selectedItem, desktopinputRef, mobileInputRef, setSearchBarVisible, isSearchBarVisible } = useSelectedItem();
  useEffect(() => {
    const handleResize = () => {
      if (window.innerWidth >= 640) {
        setDropdownOpen(false);
        setSearchBarVisible(false)
      }
    };
    window.addEventListener('resize', handleResize);
    handleResize();
    return () => {
      window.removeEventListener('resize', handleResize);
    };
  }, []);
  const handleClick = () => {
    setSearchBarVisible(true);
    if (window.innerWidth >= 640 && desktopinputRef) {
      desktopinputRef.current.focus();
    } else if (mobileInputRef.current) {
      mobileInputRef.current.focus();
    }
  };

  return (
    <div className='main'>
      <div className='fixed w-full z-20'>
        <div className='title bg-red-700 text-white text-center p-1 font-bold' style={{ backgroundColor: selectedItem ? selectedItem?.backgroundColor : "" }}>
          EVERY OUTFIT HAS A LOVE STORY - LET'S CREATE YOURS TOGETHER!
        </div>
        <div className='flex sm:hidden text-white justify-between items-center h-14 px-3' style={{ backgroundColor: selectedItem ? selectedItem?.headerBackground : "black" }}>
          <div className='flex items-center'>
            <button  data-testid="drop-down" onClick={() => setDropdownOpen(!isDropdownOpen)}>
              <i className="fas fa-bars"></i>
            </button>
          </div>
          {!isSearchBarVisible ? (
            <Link to={"/"} >
              <div className='text-xl italic font-normal'>
                <p>Swiftly Styled</p>
              </div>
            </Link>
          ) : (
            <div className='border relative flex justify-center items-center rounded-full bg-white text-black w-11/12 h-10 mx-3'>
              <div className='w-full flex h-8 items-center pl-3'>
                <input
                  type="text"
                  placeholder='Search for AI-powered inspiration...'
                  className='outline-none w-full h-full px-3 text-xs md:text-sm'
                  id='searchInput'
                  ref={mobileInputRef}
                />
              </div>
              <div className='flex items-center gap-2 md:gap-2.5 px-3 text-gray-600'>
                <div>|</div>
                <i data-testid="microphone-icon" className="fas fa-microphone"></i>
                <i className="far fa-image"></i>
                <i  className="fas fa-search"></i>
              </div>
            </div>
          )}

          <div className='flex items-center'>
            <button  data-testid="searchVisible" onClick={() => setSearchBarVisible(!isSearchBarVisible)}>
              {isSearchBarVisible ? <i className="fa-solid fa-xmark"></i> : <i className="fas fa-search"></i>}
            </button>
            <button className='ml-4'>
              <i className="far fa-user"></i>
            </button>
          </div>
        </div>
        {isDropdownOpen && (
          <div className='bg-gray-200 text-gray-600 font-semibold p-4 sm:hidden' onClick={() => {
            handleCollection(true);
            setDropdownOpen(false);
          }}>
            Collections
          </div>
        )}

        <div className='sm:flex justify-around hidden items-center md:px-16 h-14' style={{ backgroundColor: selectedItem ? selectedItem?.headerBackground : "black" }}>
          <Link to={"/"} >
            <div className='text-white italic font-normal text-lg lg:text-2xl'>
              Swiftly Styled
            </div>
          </Link>
          <div className='border relative flex justify-center items-center rounded-full bg-white md:w-3/6 h-10'>
            <div className='w-full flex h-full items-center pl-3'>
              <input
                type="text"
                placeholder='Search for AI-powered inspiration...'
                className='outline-none w-full h-8 px-3 text-xs md:text-sm'
                id='searchInput'
                ref={desktopinputRef}
              />
            </div>
            <div className='flex items-center gap-2 md:gap-2.5 px-3 text-gray-600'>
              <div>|</div>
              <i className="fas fa-microphone"></i>
              <i className="far fa-image"></i>
              <i className="fas fa-search" data-testid="search" onClick={handleClick}></i>
            </div>
          </div>
          <div className='text-white font-bold text-xs'  data-testid="collection" onClick={handleCollection}>
            COLLECTIONS
          </div>
          <div className='text-white'>
            <i className="far fa-user"></i>
          </div>
        </div>
      </div>
    </div>

  );
};

export default Header;
