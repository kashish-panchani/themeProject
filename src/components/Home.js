import React, { useEffect, useState } from 'react'
import { Link } from 'react-router-dom'
import { cardItems, theme } from '../constants/home'
import { useSelectedItem } from './SelectedItemContext';
import { handleCollection } from '../utils/utils';

const Home = () => {
    const { setSelectedItem, desktopinputRef, mobileInputRef, setSearchBarVisible, isSearchBarVisible } = useSelectedItem();
    const [totalItems, setTotalItems] = useState([]);
    useEffect(() => {
        setSelectedItem(null);
    }, [setSelectedItem]);

    useEffect(() => {
        window.scrollTo(0, 0);
    }, []);

    const styles = {
        card: {
            margin: "2px",
            padding: "0",
            position: "relative",
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
            overflow: "hidden",
        },
        s: {
            gridRowEnd: "span 10",
        },
        l: {
            gridRowEnd: "span 15"
        }
    }

    useEffect(() => {
        setTotalItems(() => {
            let array = cardItems?.map((item, i) => i + 1);
            return (array.slice(-2).concat(array.slice(0, -2)))
        });
    }, [cardItems]);

    useEffect(() => {
        if (isSearchBarVisible) {
            if (window.innerWidth >= 640 && desktopinputRef) {
                desktopinputRef.current.focus();
            } else if (mobileInputRef.current) {
                mobileInputRef.current.focus();
            }
        }
    }, [isSearchBarVisible, desktopinputRef, mobileInputRef]);

    const handleClick = () => {
        setSearchBarVisible(true);
        if (window.innerWidth >= 640 && desktopinputRef) {
            desktopinputRef.current.focus();
        } else if (mobileInputRef.current) {
            mobileInputRef.current.focus();
        }
    };

    const handlePrevBtn = () => {
        setTotalItems((prevItems) => {
            let newItems = [...prevItems];
            newItems.unshift(newItems.pop());
            return newItems;
        });

    };

    const handleNextBtn = () => {
        setTotalItems((prevItems) => {
            let newItems = [...prevItems];
            newItems.push(newItems.shift());
            return newItems;
        });
    };
    return (
        <>
            <div className='overflow-hidden'>
                <div className='uppercase my-6 md:my-10 flex flex-col justify-center items-center text-center italic font-black text-base md:text-5xl text-slate-700 w-full'>
                    <span>perfect outfits<br /> from across all eras</span>
                </div>
                <div className='text-center italic text-base md:text-2xl'>
                    <p className='text-slate-700'>
                        <span className='font-black capitalize'> Swiftly Style </span> your outfit for the <span className='font-black capitalize'>Eras Tour!</span>
                    </p>
                </div>
                <div className='text-center text-sm md:text-lg mt-8 md:mt-14 font-normal text-gray-600'>
                    <p>
                        Got a <span className='italic'>Blank Space </span>in your wardrobe?
                    </p>
                    <p>
                        Discover <span>Enchanted</span>fashion ideas beyond your Wildest Dreams!
                    </p>
                </div>
                <div className='mt-8 flex justify-center items-center font-medium gap-2'>
                    <button
                        className='btn border w-36 py-2 uppercase bg-rose-600 font-bold text-white'
                        onClick={handleCollection}
                    >
                        explore collections
                    </button>
                    <button className='btn border-black border w-36 font-bold uppercase py-2' onClick={handleClick} >
                        search outfits
                    </button>
                </div>
                <div className='mt-8 md:mt-16 text-center text-xs md:text-sm text-gray-600'>
                    <p>KARMA IS YOUR FLAWLESS ATTIRE.</p>
                </div>
                <div className='flex mt-2'>
                    {theme.map((el, index) => (
                        <div key={index} style={{
                            backgroundColor: el.themeColor,
                        }} className='w-full h-7 sm:h-10 lg:h-14'></div>
                    ))}
                </div>
                <div id='collectionID'>
                    <div className='text-slate-800 flex flex-col gap-2 justify-center items-center my-6' >
                        <p className='text-base md:text-4xl flex justify-center items-center italic font-black' >Collections from the Eras!</p>
                        <p className='w-80 md:w-96 text-xs md:text-sm font-normal text-center text-gray-600'>Browse through our Eras Tour inspired collections,<br />featuring fashion from leading ecommerce platforms.
                        </p>
                    </div>
                    <div className='pb-2'>
                        <div
                            style={{
                                gridAutoRows: '14px',
                            }} className='mx-auto grid gap-1 max-w-22 md:max-w-xl grid-cols-2 md:grid-cols-3'>
                            {cardItems.map((el) => (
                                <Link key={el.id} to={`/collection/${el.text[0]}`} style={{
                                    ...styles.card,
                                    ...styles[el.size],
                                    backgroundColor: el.backgroundColor,
                                    height: el.height
                                }}
                                    className='card'
                                >
                                    <img src={el.icon} alt={el.alt} className="image mb-5 absolute flex justify-center items-center opacity-30" />
                                    <div className='text-center'>
                                        {el?.text?.map((e, index) => (
                                            <span key={index} className="overlay-text hidden" style={{ fontFamily: el.fontFamily }}>{e}</span>
                                        ))}
                                    </div>
                                </Link>
                            ))}
                        </div>
                    </div>
                </div>
                <div className='overflow-hidden'>
                    <div className="w-full py-7 px-11">
                        <div className='text-base md:text-4xl text-center italic font-black'>Featured Collections</div>
                        <div className='text-center pb-10'>Check out the most popular collections on our website right now!</div>
                        <div className="flex justify-center items-center my-0 mx-auto relative h-80" style={{ maxWidth: "1200px" }}>
                            {cardItems.map((item, i) => {
                                const items = [...totalItems].splice(0, 5);
                                return (
                                    <div
                                        key={i}
                                        className={`allItems ${items.includes(i + 1)
                                            ? `allItems_${items.indexOf(i + 1) + 1}`
                                            : ""
                                            }`}
                                    >
                                        <img
                                            className="w-full h-full object-cover rounded-xl"
                                            src={item.sliderImg}
                                            alt={item.alt}
                                            draggable={false}
                                        />
                                    </div>
                                );
                            })}
                            <div
                                data-testid="prevbtn"
                                className='flex justify-center items-center absolute left-16 m-7 ani-left'
                                onClick={handlePrevBtn}
                            >
                                <div className='flex justify-center items-center w-14 h-14 bg-black rounded-full'>
                                    <i class="fa-solid fa-arrow-left text-white"></i>
                                </div>
                            </div>
                            <div
                                data-testid="nextbtn"
                                className='flex justify-center items-center absolute right-16 m-7 ani-left'
                                onClick={handleNextBtn}
                            >
                                <div className='flex justify-center items-center w-14 h-14 bg-black rounded-full'>
                                    <i class="fa-solid fa-arrow-right text-white"></i>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <div className='flex mt-2'>
                    {theme.map((el, index) => (
                        <div key={index} style={{
                            backgroundColor: el.themeColor,
                        }} className='w-full h-6 sm:h-10 lg:h-14'></div>
                    ))}
                </div>
                <div className='footer_bg text-white flex flex-col gap-3 items-center  justify-center text-[10px] md:text-base py-3'>
                    <p className='font-normal text-center'>THIS SITE IS OPERATED BY<span className='font-bold'>SWIFTIES </span> (WHO <span className='italic'>WERE</span> ALSO STRUGGLING TO FIND OUTFITS FOR THE ERAS TOUR).</p>
                    <p className='text-center'>WE WOULD LIKE TO CLARIFY THAT <span className='font-bold'> WE ARE NOT AFFILIATED</span> WITH TAYLOR SWIFT, TAYLOR NATION, OR THE ERAS TOUR.</p>
                    <p className='text-center'>ALL IMAGES FEATURED ON THIS WEBSITE ARE USED FOR ILLUSTRATIVE PURPOSES ONLY <br /> AND WE DO NOT CLAIM OWNERSHIP OF THEM.</p>
                </div>
                <div className='footer_bg2 text-white flex flex-col gap-3  items-center justify-center text-[10px] md:text-base py-3'>
                    <p className='font-normal text-center'>This website features links to products from our affiliate partners <br /> to provide a means for us to earn commissions, if you choose to make a purchase through our links (at no added cost to you!)</p>
                    <p className='text-center'>We graciously thank you for helping us pay off the loan we had to take to purchase our tickets for the Eras Tour.</p>
                </div>
                <div className='footer_bg3 text-white flex justify-between items-center px-3 md:px-24 py-1 md:py-2 text-[9px] md:text-xs'>
                    <div className='font-bold'>PRIVACY</div>
                    <div className='text-center'>2024 SwiftlyStyled. All Rights Reserved.<br />London,UK</div>
                    <div className='font-bold'>TERMS</div>
                </div>
            </div>
        </>
    );
}

export default Home;
