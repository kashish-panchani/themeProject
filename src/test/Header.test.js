import { fireEvent, render, screen } from "@testing-library/react";
import { SelectedItemProvider } from "../components/SelectedItemContext";
import Header from "../components/Header";
import { BrowserRouter } from "react-router-dom";

test("testing", () => {
  render(
    <BrowserRouter>
      <SelectedItemProvider>
        <Header />
      </SelectedItemProvider>
    </BrowserRouter>
  )
  const mainTitle = screen.getByText("EVERY OUTFIT HAS A LOVE STORY - LET'S CREATE YOURS TOGETHER!")
  expect(mainTitle).toBeInTheDocument();
  const label = screen.getByText(/Collections/i);
  expect(label).toBeInTheDocument();
  const logo = screen.getAllByText(/Swiftly Styled/i)
  expect(logo.length).toBe(2);
  const input = screen.getByRole('textbox')
  expect(input).toBeInTheDocument();
  const searchInput = screen.getByPlaceholderText('Search for AI-powered inspiration...');
  expect(searchInput).toBeInTheDocument();
  const search = screen.getByTestId('search');
  const handleClickMock = jest.fn();
  search.onclick = handleClickMock;
  fireEvent.click(search);
  expect(handleClickMock).toHaveBeenCalled();
  const collection = screen.getByTestId('collection');
  const handleCollectionMock = jest.fn();
  collection.onclick = handleCollectionMock;
  fireEvent.click(collection);
  expect(handleCollectionMock).toHaveBeenCalled();
  const dropDown = screen.getByTestId('drop-down');
  const setDropdownOpenMock = jest.fn();
  dropDown.onclick = setDropdownOpenMock;
  fireEvent.click(dropDown);
  expect(setDropdownOpenMock).toHaveBeenCalled();
  const searchVisible = screen.getByTestId('searchVisible');
  const setSearchBarVisibleMock = jest.fn();
  searchVisible.onclick = setSearchBarVisibleMock;
  fireEvent.click(searchVisible);
  expect(setSearchBarVisibleMock).toHaveBeenCalled();
})
test('snapshot', () => {
  const { asFragment } = render(
    <BrowserRouter>
    <SelectedItemProvider>
      <Header />
    </SelectedItemProvider>
  </BrowserRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});