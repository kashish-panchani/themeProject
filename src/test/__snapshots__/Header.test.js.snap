// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="fixed w-full z-20"
    >
      <div
        class="title bg-red-700 text-white text-center p-1 font-bold"
      >
        EVERY OUTFIT HAS A LOVE STORY - LET'S CREATE YOURS TOGETHER!
      </div>
      <div
        class="flex sm:hidden text-white justify-between items-center h-14 px-3"
      >
        <div
          class="flex items-center"
        >
          <button
            data-testid="drop-down"
          >
            <i
              class="fas fa-bars"
            />
          </button>
        </div>
        <a
          href="/"
        >
          <div
            class="text-xl italic font-normal"
          >
            <p>
              Swiftly Styled
            </p>
          </div>
        </a>
        <div
          class="flex items-center"
        >
          <button
            data-testid="searchVisible"
          >
            <i
              class="fas fa-search"
            />
          </button>
          <button
            class="ml-4"
          >
            <i
              class="far fa-user"
            />
          </button>
        </div>
      </div>
      <div
        class="sm:flex justify-around hidden items-center md:px-16 h-14"
      >
        <a
          href="/"
        >
          <div
            class="text-white italic font-normal text-lg lg:text-2xl"
          >
            Swiftly Styled
          </div>
        </a>
        <div
          class="border relative flex justify-center items-center rounded-full bg-white md:w-3/6 h-10"
        >
          <div
            class="w-full flex h-full items-center pl-3"
          >
            <input
              class="outline-none w-full h-8 px-3 text-xs md:text-sm"
              id="searchInput"
              placeholder="Search for AI-powered inspiration..."
              type="text"
            />
          </div>
          <div
            class="flex items-center gap-2 md:gap-2.5 px-3 text-gray-600"
          >
            <div>
              |
            </div>
            <i
              class="fas fa-microphone"
            />
            <i
              class="far fa-image"
            />
            <i
              class="fas fa-search"
              data-testid="search"
            />
          </div>
        </div>
        <div
          class="text-white font-bold text-xs"
          data-testid="collection"
        >
          COLLECTIONS
        </div>
        <div
          class="text-white"
        >
          <i
            class="far fa-user"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
